src/Protocol/Diaspora/Repository/DiasporaContact.php

```diff
@@ -0,0 +1,260 @@
+
+		$fields = [
+			'uri-id'            => $uriId,
+			'addr'              => $DiasporaContact->addr,
+			'alias'             => (string)$DiasporaContact->alias,
+			'nick'              => $DiasporaContact->nick,
+			'name'              => $DiasporaContact->name,
+			'given-name'        => $DiasporaContact->givenName,
+			'family-name'       => $DiasporaContact->familyName,
+			'photo'             => (string)$DiasporaContact->photo,
+			'photo-medium'      => (string)$DiasporaContact->photoMedium,
+			'photo-small'       => (string)$DiasporaContact->photoSmall,
+			'batch'             => (string)$DiasporaContact->batch,
+			'notify'            => (string)$DiasporaContact->notify,
+			'poll'              => (string)$DiasporaContact->poll,
+			'subscribe'         => (string)$DiasporaContact->subscribe,
+			'searchable'        => $DiasporaContact->searchable,
+			'pubKey'            => $DiasporaContact->pubKey,
+			'baseurl'           => (string)$DiasporaContact->baseurl,
+			'gsid'              => $DiasporaContact->gsid,
+			'updated'           => DateTimeFormat::utcNow(),
+			'interacting_count' => $DiasporaContact->interacting_count,
+			'interacted_count'  => $DiasporaContact->interacted_count,
+			'post_count'        => $DiasporaContact->post_count,
+		];
+
+		// Limit the length on incoming fields
+		$fields = $this->definition->truncateFieldsForTable('diaspora-contact', $fields);
+
+		$this->db->insert('diaspora-contact', $fields, Database::INSERT_UPDATE);
+
+		return $this->selectOneByUriId($uriId);
+	}
+
+	/**
+	 * Fetches a Diaspora profile from a given URI and updates it depending on the mode
+	 *
+	 * @param WebFingerUri $uri    profile URI
+	 * @param boolean      $update true = always update, false = never update, null = update when not found or outdated
+	 * @return Entity\DiasporaContact
+	 * @throws HTTPException\NotFoundException
+	 */
+	public function getByAddr(WebFingerUri $uri, ?bool $update = self::UPDATE_IF_MISSING_OR_OUTDATED): Entity\DiasporaContact
+	{
+		if ($update !== self::ALWAYS_UPDATE) {
+			try {
+				$dcontact = $this->selectOneByAddr($uri);
+				if ($update === self::NEVER_UPDATE) {
+					return $dcontact;
+				}
+			} catch (HTTPException\NotFoundException $e) {
+				if ($update === self::NEVER_UPDATE) {
+					throw $e;
+				}
+			}
+		}
+
+		// Trigger a contact update if needed
+		$contact = Contact::getByURL($uri, $update, ['uri-id', 'network', 'next-update']);
+		if (empty($contact['uri-id'])) {
+			throw new HTTPException\NotFoundException('Diaspora profile with URI ' . $uri . ' not found');
+		}
+
+		return self::selectOneByUriId($contact['uri-id']);
+	}
+
+	/**
+	 * Update or create a diaspora-contact entry via a probe array
+	 *
+	 * @param array $data Probe array
+	 * @return Entity\DiasporaContact
+	 * @throws \Exception
+	 */
+	public function updateFromProbeArray(array $data): Entity\DiasporaContact
+	{
+		$uriId = ItemURI::insert(['uri' => $data['url'], 'guid' => $data['guid']]);
```

Comment:
Isn't this redundant to the functionality in the `save` function?

Reply:
You're right, I've been flip-flopping about where to put this and when I finally put it in the `save` method I didn't remove it from here.

Reply:
Yes, but we still need it here to retrieve the follow/er/s counts.

Reply:
It's already the case, see line 124.